soh/src/code/z_parameter.c

```diff
@@ -1618,7 +1618,9 @@ void func_80084BF4(PlayState* play, u16 flag) {
 // (special cases for some duplicate items)
 void GameplayStats_SetTimestamp(u8 item) {
 
-    if (gSaveContext.sohStats.timestamp[item] != 0) {
+    // If the item already has a timestamp or if it's a boss key, do nothing
+    // ITEM_KEY_BOSS is used to timestamp Ganon's boss key. Handled in z_play.c
```

Comment:
why not just check for it being ganon's key here? it seems you could just do

```c
if (gSaveContext.sohStats.timestamp[item] != 0 ||
    (item == ITEM_KEY_BOSS && !(gSaveContext.inventory.dungeonItems[10] & 1)) {
    return;
}
```

Reply:
Unfortunately this method didn't work since the inventory gets updated after this function gets called, so you end up with no timestamp. However, what did work is checking for `RG_GANONS_CASTLE_BOSS_KEY` a little further down in `Randomizer_GameplayStats_SetTimestamp()` - we just then have to add a few more lines to check for it in the case where it's not keysanity or in the vanilla case.

Reply:
Ended up being more simple than I expected. Gave it the changes some testing in vanilla & rando with boss keys shuffled and seems to be working. Definitely better than having it in z_play